src/nmark/parseInline.nim

```diff
@@ -632,34 +622,32 @@ proc processEmphasis*(line: string): seq[DelimStack] =
 
 
 proc parseInline*(line: string): seq[DelimStack] =
-
-  var r = (line.readAutoLink &
-   line.readLinkOrImage &
-   line.readCodeSpan &
-   line.readEmphasisAste &
-   line.readEmphasisUnder &
-   line.readHardBreak &
-   line.readEntity &
-   line.readEscape)
-   .sortedByIt(it.position)
-
+  # r
+  result = line.readAutoLink()
+  result.add line.readLinkOrImage
+  result.add line.readCodeSpan
+  result.add line.readEmphasisAste
+  result.add line.readEmphasisUnder
+  result.add line.readHardBreak
+  result.add line.readEntity
+  result.add line.readEscape
+  result.sort(proc (a, b: DelimStack): int = cmp(a.position, b.position))
   #echoObj r
 
-  let n_em = r.parseEscape
-              .parseAutoLink(line)
-              .parseCodeSpan.parseLink(line)
-              .filter(proc(x: DelimStack): bool =
-              (x.typeDelim != "*" and x.typeDelim != "_"))
-
+  # n_em 
+  result.parseEscape()
+  result.parseAutoLink(line)
+  result.parseCodeSpan()
+  result.parseLink(line)
+  result.keepItIf(it.typeDelim notin ["*", "_"])
+  
   #echoObj n_em
   #echoObj r
-  let em = r.parseEmphasis
+  let em = result.parseEmphasis()
 
   #echoObj em
   #echoObj (n_em & em)
 
-  return (n_em & em)
-         .sortedByIt(it.position)
-         .filter(proc(x: DelimStack): bool =
-         (x.isActive) and
-          x.potential != both)
+  result.add em
+  result.sort(proc (a, b: DelimStack): int = cmp(a.position, b.position))
+  result.keepItIf(it.isActive and it.potential != both)
```

Comment:
So, this is one of the "bigger" changes. Basically, I made most operations here work in-place instead of out-of-place, so they create fewer sequences. I followed the original algorithm as close as I can, and the tests pass, but please check if the code is wrong or not.